File: BOJ/Gold IV/BOJ1806.py
#BOJ1806 부분합 20210328
import sys
from collections import deque
input = sys.stdin.readline

def main():
    n,s = map(int,input().rstrip().split())
    li = list(map(int,input().rstrip().split()))
    sum_ = 0
    q = deque()
    mn = 100001
    for v in li:
        sum_ += v
        q.append(v)
        if sum_ >= s:
            while sum_ - q[0] >= s:
                sum_ -= q.popleft()
            mn = min(mn, len(q))
    print(mn if mn != 100001 else 0)
if __name__ == '__main__':
    main()

File: BOJ/Gold IV/BOJ1806.java
//BOJ1806 부분합 20210706
import java.io.*;
import java.util.*;

import static java.lang.Math.*;

public class Main {
    private static final BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    private static final BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

    public static void main(String[] args) throws IOException {
        StringTokenizer stk = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(stk.nextToken());
        int s = Integer.parseInt(stk.nextToken());
        int[] arr = Arrays.stream(br.readLine().split(" ")).mapToInt(Integer::parseInt).toArray();

        int ans = Integer.MAX_VALUE;

        int l = 0, r = 0;
        int sum = 0;
        while(true){
            if(r == n && sum < s) break;
            if(sum < s){
                sum += arr[r];
                r++;
            }
            else{
                sum -= arr[l];
                l++;
                ans = min(r-l+1, ans);
            }
        }
        System.out.println(((ans == Integer.MAX_VALUE) ? 0 : ans));
    }
}